src/eredis_cluster_pool_worker.erl

```diff
@@ -38,6 +38,8 @@ init(Args) ->
 
     {ok, #state{conn=Conn}}.
 
+query(full, _Commands) ->
+    {error, retry};
```

Comment:
Actually, `eredis_cluster` is using `poolboy:transaction/2` which is using `poolboy:checkout/3` with Block = true. Checkout can only return `full` when Block =:= false. Therefore, if the pool is full, the transaction fun never gets called so this PR just adds dead code.

Instead, when the pool doesn't have workers available, we get an `exit:{timeout, {gen_server, call, _}}` exception from poolboy:transaction/3. This is turned into `{error, no_connection}` by the try-catch in eredis_cluster_pool and then it triggers a refesh mappings and then a retry in `eredis_cluster`.

Refresh mappings isn't necessary in this case. A retry without refresh mappings would suffice. We have done it in the Nordix fork.